pkg/apis/monitoring/v1/prometheus_types.go

```diff
@@ -345,6 +355,13 @@ type CommonPrometheusFields struct {
 	HostNetwork bool `json:"hostNetwork,omitempty"`
 	// PodTargetLabels are added to all Pod/ServiceMonitors' podTargetLabels
 	PodTargetLabels []string `json:"podTargetLabels,omitempty"`
+	// Enable access to prometheus web admin API. Defaults to the value of `false`.
+	// WARNING: Enabling the admin APIs enables mutating endpoints, to delete data,
+	// shutdown Prometheus, and more. Enabling this should be done with care and the
+	// user is advised to add additional authentication authorization via a proxy to
+	// ensure only clients authorized to perform these actions can do so.
+	// For more information see https://prometheus.io/docs/prometheus/latest/querying/api/#tsdb-admin-apis
+	EnableAdminAPI bool `json:"enableAdminAPI,omitempty"`
```

Comment:
I tried using the admin API in Prometheus agent mode but it fails with `{"status":"error","errorType":"execution","error":"unavailable with Prometheus Agent"}`. I guess there's no point in moving the field to the common struct.

Reply:
I tested starting the Prometheus Agent with the flag, which worked, but I haven't tried to actually use the endpoint. Good thing that you tried 😛